### ISSUE TITLE
1.0.2: 3 issues with "Export chat" functionnality

### ISSUE THREAD

**Issue by ageru** [redsolution/xabber-android#420]
# TL;DR
"Export chat" basically works, but there are 2 issues that break Xabber along the way, and in the end some content is not saved properly.

# Device
* [HTD Desire HD] (http://www.gsmarena.com/htc_desire_hd-3468.php)
* [CyanogenMod 11 M12] (http://forum.xda-developers.com/showthread.php?t=2533007)

# Steps to reproduce
1. Make sure you are logging chats on your device  
2. Update Xabber from 0.9.30f to 1.0.2 via F-droid
3. Tap on the name of a user to access the chat window
4. Call up the option menu
5. Select Export Chat

# Issue 1: cut/paste menu design issue
## Actual result
When selecting text to cut or paste, the cut/paste menu comes down, but all the icons are white on a white background. They are still functionnal if you know where to press.

## Expected result
When selecting text to cut or paste, the cut/paste menu comes down, icons should be a different color from the background they are on

# Issue 2: Xabber crashes upon exporting
## Actual result
When confirming the name of the export, an Android window appears to inform Xabber has crashed. If you press OK quickly, the save is not carried out. If you wait 10 to 15 secondes before dismissing the window, the chat is properly saved.

## Expected result
Xabber should not crash upon exporting a chat.

# Issue 3: Issues with parsing & conversion to HTML
## Actual result
newlines in messages are exported as simple spaces.
"<" and any character following it in a message are not exported. Ie, when I text my girlfriend "I <3 sexy babe" (don't judge me), only "I " is exported.

## Expected result
newlines in messages should be exported as `<br />` like in 0.9.30.
"<" should be exported as `&lt;` like in 0.9.30, and the text following it should be exported.

# The final word
Apologies for being lazy and grouping all this in one bug. I hope it's helpful nonetheless.

**Comment by aelmahmoudy** [comment]
Here's the logcat when Xabber crashes at chat export:
```
D/AndroidRuntime( 7710): Shutting down VM
W/dalvikvm( 7710): threadid=1: thread exiting with uncaught exception (group=0x41749ba8)
E/AndroidRuntime( 7710): FATAL EXCEPTION: main
E/AndroidRuntime( 7710): Process: com.xabber.androiddev, PID: 7710
E/AndroidRuntime( 7710): java.lang.NullPointerException
E/AndroidRuntime( 7710):        at android.widget.Toast.makeText(Toast.java:268)
E/AndroidRuntime( 7710):        at com.xabber.android.ui.dialog.ChatExportDialogFragment$ChatExportAsyncTask.onPostExecute(ChatExportDialogFragment.java:121)
E/AndroidRuntime( 7710):        at com.xabber.android.ui.dialog.ChatExportDialogFragment$ChatExportAsyncTask.onPostExecute(ChatExportDialogFragment.java:82)
E/AndroidRuntime( 7710):        at android.os.AsyncTask.finish(AsyncTask.java:632)
E/AndroidRuntime( 7710):        at android.os.AsyncTask.access$600(AsyncTask.java:177)
E/AndroidRuntime( 7710):        at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:645)
E/AndroidRuntime( 7710):        at android.os.Handler.dispatchMessage(Handler.java:102)
E/AndroidRuntime( 7710):        at android.os.Looper.loop(Looper.java:136)
E/AndroidRuntime( 7710):        at android.app.ActivityThread.main(ActivityThread.java:5001)
E/AndroidRuntime( 7710):        at java.lang.reflect.Method.invokeNative(Native Method)
E/AndroidRuntime( 7710):        at java.lang.reflect.Method.invoke(Method.java:515)
E/AndroidRuntime( 7710):        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:785)
E/AndroidRuntime( 7710):        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:601)
E/AndroidRuntime( 7710):        at dalvik.system.NativeStart.main(Native Method)
I/dalvikvm( 7710): Jit: resizing JitTable from 8192 to 16384
```

**Comment by aelmahmoudy** [comment]
I can't reproduce first issue in v1.0.5

**Comment by aelmahmoudy** [comment]
Thirst issue is probably fixed by https://github.com/redsolution/xabber-android/commits/master

**Comment by grigoryfedorov** [comment]
This a224f08a7ae44f5d0333b7035d4d4c4642a8d344 should fix crash on chat export.

**Comment by grigoryfedorov** [comment]
@ageru please, check latest version. 1.0.7 is available.

**Comment by ageru** [comment]
Hi Grigory,

My apologies for the delay answering (busy month...), and thanks for both your patience and your work.

I have just installed 1.0.12 from F-droid, here is what I found out.

Issue 1:
This is still happening, as you can see on the top of this screenshot, where the cut/copy/paste icons should be visible, but are blank. You can still access the functionnalities if you tap in the right place though.
![screenshot_2015-06-06-15-16-52](https://cloud.githubusercontent.com/assets/12240432/8020017/9b1e34ea-0c61-11e5-8b05-6ed9fd1b9fb9.png)

Issue 2:
This issue is fixed. Many thanks :)

Issue 3:
This issue is fixed. Many thanks :)

**Comment by ageru** [comment]
Most important issues fixed, closing thie bug.
Will open a new bug report for the copy/paste issue
